packages/components/src/components/audio/audio.stories.ts

```diff
@@ -0,0 +1,366 @@
+
+const { argTypes, args, parameters } = storybookDefaults('sd-audio');
+const { generateTemplate } = storybookTemplate('sd-audio');
+
+export default {
+  title: 'Components/sd-audio',
+  component: 'sd-audio',
+  args,
+  argTypes,
+  parameters: { ...parameters },
+  decorators: [withActions] as any
+};
+
+const audioElementScript = html`<script>
+  const initAudioPlayer = () => {
+    const audioElements = document.querySelectorAll('#audio-example');
+    const sdAudioPlayers = document.querySelectorAll('#sd-audio-example > sd-audio');
+
+    // Select the first or second audio element based on availability
+    // Due to Storybook behavior, we need to carefully target the appropriate elements in the context of the story.
+    const audioElement =
+      document.querySelectorAll('#audio-example')[1] || document.querySelectorAll('#audio-example')[0];
+    const sdAudio =
+      document.querySelectorAll('#sd-audio-example > sd-audio')[1] ||
+      document.querySelectorAll('#sd-audio-example > sd-audio')[0];
+
+    function setupEventListeners() {
+      sdAudio?.addEventListener('sd-playback-start', playAudio);
+      sdAudio?.addEventListener('sd-playback-mute', muteAudio);
+      sdAudio?.addEventListener('sd-playback-speed', updatePlaybackSpeed);
+    }
+
+    const playAudio = () => {
+      sdAudio.isPlaying ? audioElement.pause() : audioElement.play();
+    };
+
+    const muteAudio = () => {
+      audioElement.muted = !sdAudio.isMuted;
+    };
+
+    const updatePlaybackSpeed = () => {
+      sdAudio.playbackSpeed = audioElement.playbackRate === 1.5 ? 1 : audioElement.playbackRate + 0.25;
+      audioElement.playbackRate = sdAudio.playbackSpeed;
+    };
+
+    function removeEventListeners() {
+      sdAudioPlayers.forEach(sdAudio => {
+        sdAudio.playing = false;
+        sdAudio?.removeEventListener('sd-playback-start', playAudio);
+        sdAudio?.removeEventListener('sd-playback-mute', muteAudio);
+        sdAudio?.removeEventListener('sd-playback-speed', updatePlaybackSpeed);
+      });
+    }
+
+    removeEventListeners();
+    setupEventListeners();
+  };
+
+  // Run the script when the page loads
+  initAudioPlayer();
+
+  // Run the script again whenever the hash (path) changes
+  window.onhashchange = initAudioPlayer;
+</script>`;
+
+const audioConstant: ConstantDefinition = {
+  type: 'slot',
+  name: 'default',
+  value: '<audio id="audio-example" slot="default" src="./placeholders/audio/sds-song.mp3" preload="metadata"></audio>'
+};
+
+export const Default = {
+  name: 'Default',
+  parameters: {
+    controls: {
+      exclude: ['playbackSpeed']
+    }
+  },
+  render: (args: any) => {
+    return html`
+      <div id="sd-audio-example" className="p-0">
+        ${generateTemplate({
+          args,
+          constants: [
+            audioConstant,
+            {
+              type: 'slot',
+              name: 'transcript',
+              value: '<div>Show audio transcript</div>'
```

Comment:
can we add our transcript here? Files should be available in our teams channel (under files > sample audio > sds-song_transcript.vtt